src/Ocelot/Configuration/RateLimitOptions.cs

```diff
@@ -18,73 +19,79 @@ public RateLimitOptions(bool enableRateLimiting, string clientIdHeader, Func<Lis
         /// </summary>
         /// <value>
         /// A <see cref="List{T}"/> collection with white listed clients.
         /// </value>
-        public List<string> ClientWhitelist => _getClientWhitelist();
-
+        public List<string> ClientWhitelist => _getClientWhitelist();
+
         /// <summary>
         /// Gets or sets the HTTP header that holds the client identifier, by default is X-ClientId.
         /// </summary>
         /// <value>
         /// A string value with the HTTP header.
         /// </value>
-        public string ClientIdHeader { get; }
-
+        public string ClientIdHeader { get; }
+
         /// <summary>
         /// Gets or sets the HTTP Status code returned when rate limiting occurs, by default value is set to 429 (Too Many Requests).
         /// </summary>
         /// <value>
-        /// An integer value with the HTTP Status code.
-        /// <para>Default value: 429 (Too Many Requests).</para>
+        /// An integer value with the HTTP Status code.
+        /// <para>Default value: 429 (Too Many Requests).</para>
         /// </value>
-        public int HttpStatusCode { get; }
-
+        public int HttpStatusCode { get; }
+
         /// <summary>
-        /// Gets or sets a value that will be used as a formatter for the QuotaExceeded response message.
+        /// Gets or sets a value that will be used as a formatter for the QuotaExceeded response message.
         /// <para>If none specified the default will be: "API calls quota exceeded! maximum admitted {0} per {1}".</para>
         /// </summary>
         /// <value>
         /// A string value with a formatter for the QuotaExceeded response message.
         /// <para>Default will be: "API calls quota exceeded! maximum admitted {0} per {1}".</para>
         /// </value>
-        public string QuotaExceededMessage { get; }
-
+        public string QuotaExceededMessage { get; }
+
         /// <summary>
         /// Gets or sets the counter prefix, used to compose the rate limit counter cache key.
         /// </summary>
         /// <value>
         /// A string value with the counter prefix.
         /// </value>
-        public string RateLimitCounterPrefix { get; }
-
+        public string RateLimitCounterPrefix { get; }
+
         /// <summary>
         /// Enables endpoint rate limiting based URL path and HTTP verb.
         /// </summary>
         /// <value>
         /// A boolean value for enabling endpoint rate limiting based URL path and HTTP verb.
         /// </value>
-        public bool EnableRateLimiting { get; }
-
+        public bool EnableRateLimiting { get; }
+
         /// <summary>
         /// Disables X-Rate-Limit and Rety-After headers.
         /// </summary>
         /// <value>
         /// A boolean value for disabling X-Rate-Limit and Rety-After headers.
         /// </value>
         public bool DisableRateLimitHeaders { get; }
+
+        public RateLimitMiddlewareType RateLimitMiddlewareType { get; }

```

Comment:
I'm quite certain this is overengineering. We can simply determine the `PolicyName` value to be either empty or non-empty.
Let me to explain... If we had more than 3 middleware types, then defining an enum would be necessary. However, since we only have 2 types (Ocelot and AspNet), the enumeration is not required. So, we can decide on policy value (defined or not defined).